php/libraries/DefaultFilesUploadHandler.class.inc

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Default class for handling file uploads in LORIS
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * Provides a default handler for files that were uploaded on a LORIS
+ * web site using LorisForm.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class DefaultFilesUploadHandler implements FilesUploadHandler
+{
+    /**
+     * Default directory to move the uploaded files to if none is explicitely
+     * specified.
+     */
+    public static $DEFAULT_MOVE_DIR = __DIR__ . "../user_uploads";
+    /**
+     *  The LorisForm used to upload the file(s).
+     */
+    private $_lorisForm;
```

Comment:
This adds unnecessary coupling to LorisForm.

The places where it is used, it would make more sense to use a PSR7 ServerRequestInterface to get the files out of.

Reply:
The coupling is intentional. The purpose of the entire class is to provide default behaviour for entities that handle files uploaded via a LorisForm (which is what will be the case in LORIS). I think the coupling had to be avoided in the interface, so that you could have a concrete class that implements it but still uploads files via a mechanism other than LorisForm.